src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,127 @@
+package mate.academy.dao;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.ConnectionUtil;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (
+                Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setDouble(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new RuntimeException(
+                        "Expected to insert at least one row, but inserted 0 rows.");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                BigInteger id = generatedKeys.getObject(1, BigInteger.class);
+                book.setId(id);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't create a book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (
+                Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            statement.executeQuery();
+            ResultSet resultSet = statement.getResultSet();
+            if (resultSet.next()) {
+                Book book = new Book();
+                book.setId(resultSet.getObject("id", BigInteger.class));
+                book.setTitle(resultSet.getObject("title", String.class));
+                book.setPrice(resultSet.getObject("price", Double.class));
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

Comment:
```suggestion
            throw new DataProcessingException(e);
```
Check other places. 
Aslo, provide informative message